app/services/fact_importer.rb

```diff
@@ -2,20 +2,6 @@ class FactImporter
   delegate :logger, :to => :Rails
   attr_reader :counters
 
-  def self.importer_for(type)
-    importers[type.to_s]
-  end
-
-  def self.importers
-    @importers ||= {}.with_indifferent_access
-  end
-
-  def self.register_fact_importer(key, klass, default = false)
-    importers.default = klass if default
-
-    importers[key.to_sym] = klass
-  end
-
   def self.fact_features
     importers.map { |_type, importer| importer.authorized_smart_proxy_features }.compact.flatten.uniq
```

Comment:
This method will not work, it should be either part of the registry, or reference the registry correctly.

@stbenjam, I am not familiar with Smart Proxy Authorization, what is your opinion? Is it a part of the registry, or just a service method that should stay here?

Reply:
Probably part of the registry.  The hosts controller needs to know the smart proxy features for all of the registered importers.